src/pages/Search/__test__/Search.test.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { Route, MemoryRouter } from 'react-router-dom';
+import {
+  render,
+  cleanup,
+  waitForElementToBeRemoved,
+  fireEvent,
+  waitFor,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { ThemeProvider } from 'styled-components';
+import axiosMock from 'axios';
+import Search from '../index';
+import Theme from '../../../styles/theme';
+import posts from './posts.json';
+
+jest.mock('axios');
+afterEach(cleanup);
+
+const renderWithRouter = (component, subreddit) => ({
+  ...render(
+    <ThemeProvider theme={Theme}>
+      <MemoryRouter initialEntries={[`/search/${subreddit}`]}>
+        <Route path="/search/:subreddit">{component}</Route>
+      </MemoryRouter>
+    </ThemeProvider>,
+  ),
+});
+
+it('Should show javascript on search input', async () => {
+  const { getByTestId } = renderWithRouter(<Search />, 'javascript');
+
+  const input = getByTestId('search-input');
+
+  expect(input.value).toBe('javascript');
+
+  await waitForElementToBeRemoved(() => getByTestId('loader'));
+});
+
+it('Should load data and show heatmap', async () => {
+  axiosMock.get.mockResolvedValueOnce({
+    data: posts,
+  });
+
+  const { getByTestId } = renderWithRouter(<Search />, 'javascript');
+
+  const heatmap = await waitFor(() => getByTestId('heatmap'));
+
+  expect(heatmap).toHaveTextContent('All times are shown in your timezone');
+  expect(axiosMock.get).toHaveBeenCalledTimes(2);
+});
+
+it('Should update url on button click and load data', async () => {
+  axiosMock.get.mockResolvedValue({
+    data: posts,
+  });
+
+  const { getByTestId } = renderWithRouter(<Search />, 'javascript');
+
+  const input = await getByTestId('search-input');
+  fireEvent.change(input, { target: { value: 'reactjs' } });
+  expect(input.value).toBe('reactjs');
+  fireEvent.click(getByTestId('search-button'));
+
+  const heatmap = await waitFor(() => getByTestId('heatmap'));
+
+  expect(heatmap).toHaveTextContent('All times are shown in your timezone');
+  expect(axiosMock.get).toHaveBeenCalledTimes(4);
+});
+
+it('Should show table when box clicked', async () => {
```

Comment:
This test fails for me 

![image](https://user-images.githubusercontent.com/9722792/79327102-2e3c5200-7f14-11ea-921c-af26f46cd4c3.png)

Reply:
in this case, should I just check for the table if it appears because if I check for text it  others won't have the same  result as me

Reply:
Why will others but have the same results? Don't you always use the same mock data and click on the same box inside the heatmap? Then the result should be the same everywhere I think.

Anyway, the test needs to work the same everywhere because it will also run in the GitHub pipelines usually. I added it there also in addition to linting.

I think you should check the text inside the table to be sure that selecting an hour weeks correctly. Check every row in the table at best